core/src/cubos/core/gl/util.cpp

```diff
@@ -3,34 +3,30 @@
 using namespace cubos::core;
 using namespace cubos::core::gl;
 
-void gl::generateScreenQuad(RenderDevice& renderDevice, const ShaderPipeline& pipeline, VertexArray& va,
-                            IndexBuffer& ib)
+void gl::generateScreenQuad(RenderDevice& renderDevice, const ShaderPipeline& pipeline, VertexArray& va)
 {
+    // The vertices of the quad (positions + uvs)
     float screenVerts[] = {
-        -1.0f, -1.0f, 0.0f, 0.0f, -1.0f, +1.0f, 0.0f, 1.0f, +1.0f, +1.0f, 1.0f, 1.0f, +1.0f, -1.0f, 1.0f, 0.0f,
+        -1.0f, -1.0f, 0.0f, 0.0f, -1.0f, +1.0f, 0.0f, 1.0f, +1.0f, +1.0f, 1.0f, 1.0f,
+        +1.0f, +1.0f, 1.0f, 1.0f, +1.0f, -1.0f, 1.0f, 0.0f, -1.0f, -1.0f, 0.0f, 0.0f,
     };
-    VertexBuffer vb = renderDevice.createVertexBuffer(sizeof(screenVerts), screenVerts, gl::Usage::Static);
 
-    unsigned int screenIndices[] = {
-        0, 1, 2, 2, 3, 0,
-    };
-    ib = renderDevice.createIndexBuffer(sizeof(screenIndices), screenIndices, gl::IndexFormat::UInt, gl::Usage::Static);
-
-    VertexArrayDesc screenVADesc;
-    screenVADesc.elementCount = 2;
-    screenVADesc.elements[0].name = "position";
-    screenVADesc.elements[0].type = gl::Type::Float;
-    screenVADesc.elements[0].size = 2;
-    screenVADesc.elements[0].buffer.index = 0;
-    screenVADesc.elements[0].buffer.offset = 0;
-    screenVADesc.elements[0].buffer.stride = 4 * sizeof(float);
-    screenVADesc.elements[1].name = "uv";
-    screenVADesc.elements[1].type = gl::Type::Float;
-    screenVADesc.elements[1].size = 2;
-    screenVADesc.elements[1].buffer.index = 0;
-    screenVADesc.elements[1].buffer.offset = 2 * sizeof(float);
-    screenVADesc.elements[1].buffer.stride = 4 * sizeof(float);
-    screenVADesc.buffers[0] = vb;
-    screenVADesc.shaderPipeline = pipeline;
-    va = renderDevice.createVertexArray(screenVADesc);
+    // Create the array and vertex buffer.
+    VertexArrayDesc desc;
+    desc.elementCount = 2;
+    desc.elements[0].name = "position";
+    desc.elements[0].type = gl::Type::Float;
+    desc.elements[0].size = 2;
+    desc.elements[0].buffer.index = 0;
+    desc.elements[0].buffer.offset = 0;
+    desc.elements[0].buffer.stride = 4 * sizeof(float);
+    desc.elements[1].name = "uv";
+    desc.elements[1].type = gl::Type::Float;
+    desc.elements[1].size = 2;
+    desc.elements[1].buffer.index = 0;
+    desc.elements[1].buffer.offset = 2 * sizeof(float);
+    desc.elements[1].buffer.stride = 4 * sizeof(float);
+    desc.buffers[0] = renderDevice.createVertexBuffer(sizeof(screenVerts), screenVerts, gl::Usage::Static);
+    desc.shaderPipeline = pipeline;
+    va = renderDevice.createVertexArray(desc);
```

Comment:
Why were the indices removed?

Reply:
They were unnecessary, for such a small vertex buffer, having two extra vertices won't make any performance difference.
Having an index buffer complicates the process since we need to pass it around.